### PR TITLE
Improve test output web GUI

### PR DESCRIPTION
**Pull request by 0papen0**

### CODE REVIEW

File: tools/web/resultTree/nodes/resultNode.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
+import * as b from "bobril";
+import * as styles from "../styles";
+import * as s from "../../state";
+import * as com from "../../communication";
+import { ResultTree } from "../resultTree";
+import { TreeNode, ResultTypes } from "./treeNode";
+import { clickable } from "../clickable";
+
+export class ResultNode extends TreeNode {
+    SOT: s.SuiteOrTest;
```
Comment (Bobris) [inline tools/web/resultTree/nodes/resultNode.ts]: Even though it is shortcut, it should start with lowercase.

File: tools/web/resultTree/nodes/resultNode.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    setIsFiltered() {
+        this.isFiltered =
+            this.SOT.name.includes(ResultTree.textFilter) ||
+            (this.SOT.logs && this.SOT.logs.filter(log => log.message.includes(ResultTree.textFilter)).length > 0) ||
+            (this.SOT.failures &&
+                this.SOT.failures.filter(failure => failure.message.includes(ResultTree.textFilter)).length > 0);
+    }
+
+    stackFrameToString(stackFrame: s.StackFrame): string {
+        var functionName = stackFrame.functionName || "{anonymous}";
```
Comment (keeema) [inline tools/web/resultTree/nodes/resultNode.ts]: Use let or better const. Please fix in the rest.

File: tools/web/resultTree/nodes/resultNode.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
+import * as b from "bobril";
```
Comment (keeema) [inline tools/web/resultTree/nodes/resultNode.ts]: Again, separate to many files.

File: tools/web/index.ts [hunk 4]
```diff
@@ -1,185 +1,399 @@
...
-        { tag: "h2", children: "Bobril-build 2" },
-        b.styledDiv(<b.IBobrilChildArray>[
-            s.disconnected ? "Disconnected" : s.connected ? "Connected" : "Connecting",
-            createCheckbox("Live reload", s.liveReload, com.setLiveReload),
-            s.actions.map((a) => [" ", createActionUI(a)])]),
-        getBuildStatus(),
-        getAgentsShort(selectedAgent, i => { selectedAgent = i; b.invalidate() }),
-        selectedAgent >= 0 && getAgentDetail(s.testSvrState.agents[selectedAgent])
-    ];
+    return b.createComponent({
```
Comment (keeema) [inline tools/web/index.ts]: This call creates new component factory every time. It should be defined as a standalone component.

**Pull request merged**